### PR TITLE
refactor(pgwire): refine the interface of notice

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- support notice multiple times for a query
```
dev=> create materialized view mv as select * from t where y > now() order by x;
NOTICE:  Your session timezone is UTC. It was used in the interpretation of timestamps and dates in your query. If this is unintended, change your timezone to match that of your data's with `set timezone = [timezone]` or rewrite your query with an explicit timezone conversion, e.g. with `AT TIME ZONE`.

NOTICE:  The ORDER BY clause in the CREATE MATERIALIZED VIEW statement does not guarantee that the rows selected out of this materialized view is returned in this order.
It only indicates the physical clustering of the data, which may improve the performance of queries issued against this materialized view.

CREATE_MATERIALIZED_VIEW
```
- add interface to warning to user in optimizer

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/src/optimizer/optimizer_context.rs [hunk 1]
```diff
@@ -160,6 +165,13 @@ impl OptimizerContext {
         optimizer_trace.push("\n".to_string());
     }
 
+    pub fn warn(&self, str: impl Into<String>) {
```
Comment (chenzl25) [inline src/frontend/src/optimizer/optimizer_context.rs:168]: Any usage currently?

**Review by yuhao-su — APPROVED** [review]

File: src/frontend/src/optimizer/optimizer_context.rs (hunk 1 above)
Comment (st1page) [reply src/frontend/src/optimizer/optimizer_context.rs:168]: not yet

Comment (st1page) [reply src/frontend/src/optimizer/optimizer_context.rs:168]: [not yet](https://github.com/risingwavelabs/risingwave/issues/9582)

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**